### PR TITLE
Getting started with Node.js Paypal checkout integration

### LINKED ISSUE #1962: Integrating Paypal checkout into a Nodejs application
### Integrating Paypal checkout into a Nodejs application
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article's higher ranking.*

### Introduction paragraph (2-3 paragraphs):
PayPal Holdings, Inc. is an American company operating an online payments system in most countries that support online money transfers and serves as an electronic alternative to traditional paper methods like checks and money orders. Paypal checkout allows merchants to accept payments through PayPal, MasterCard, and visa within their websites. In the tutorial, I will explain the stepwise process of integrating Paypal checkout into a Nodejs website.

### Key takeaways: 
1. Handling requests to external resources on a Nodejs application.
2. Exposing a local server to be accessed by an external server.
3. Integrating PayPal checkout into Nodejs application.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by okeloviolet**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1962

### CODE REVIEW

File: articles/nodejs-paypal-checkout/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+const app = express();
+
+app.get('/', (req, res) => res.sendFile(__dirname + "/index.html"));
+
+
+app.listen(PORT, () => console.log(`Server Started on ${PORT}`));
+```
+To obtain the `client_id` and `client_secret`, login into your Paypal developer account [here](https://developer.paypal.com/) and create a new app as shown in the image below.
+
+![Paypal create a new app](/pay/paypal-create-app.png)
```
Comment (paulodhiambo) [inline articles/nodejs-paypal-checkout/index.md:46]: Add the correct image path i.e Paypal create a new app should be Paypal create a new app

File: articles/nodejs-paypal-checkout/index.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+#### Transaction cancellation
+When a user cancels the transaction, we need to render a response to inform the user that the transaction was successfully canceled. To handle the cancellation, add the code snippet below into the `index.js` file.
+
+```javascript
+app.get('/cancel', (req, res) => res.send('Cancelled'));
+```
+- The above route returns a `Cancelled` string to the user whenever a transaction cancellation is successful.
+  
+### Conclusion
+Now that you have learned how to integrate Paypal checkout into a Node.js application, add a success and error page to the application we created. The full source for the application can be downloaded [here]().
```
Comment (paulodhiambo) [inline articles/nodejs-paypal-checkout/index.md:187]: The link to the full source code is missing.

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@okeloviolet amazing article. Review the requested changes and ping me when done.

**Comment by okeloviolet** [comment]
@paulodhiambo I have made the requested changes.

**Comment by paulodhiambo** [comment]
@okeloviolet amazing article. @hectorkambow you can give it a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
👍  nicely done @okeloviolet 
I made some minor edits for flow - please review 👍 

cc @paulodhiambo

**Pull request merged**